pallets/pallet-bonded-coins/src/types.rs

```diff
@@ -0,0 +1,100 @@
+// todo: send help!
+use parity_scale_codec::{Decode, Encode, MaxEncodedLen};
+use scale_info::TypeInfo;
+
+#[derive(Default, Clone, Encode, Decode, PartialEq, Eq, TypeInfo, MaxEncodedLen, Debug)]
+pub struct Locks {
+	pub allow_mint: bool,
+	pub allow_burn: bool,
+	pub allow_swap: bool,
+}
+
+#[derive(Clone, Encode, Decode, PartialEq, Eq, TypeInfo, MaxEncodedLen, Debug)]
+pub enum PoolStatus<LockType> {
+	Active,
+	Locked(LockType),
+	Destroying,
+}
+
+impl<LockType> PoolStatus<LockType> {
+	pub fn is_active(&self) -> bool {
+		matches!(self, Self::Active)
+	}
+
+	pub fn is_destroying(&self) -> bool {
+		matches!(self, Self::Destroying)
+	}
+
+	pub fn freeze(&mut self, lock: LockType) {
+		*self = Self::Locked(lock);
+	}
+
+	pub fn destroy(&mut self) {
+		*self = Self::Destroying;
+	}
+}
+
+#[derive(Clone, Encode, Decode, PartialEq, Eq, TypeInfo, MaxEncodedLen)]
+pub struct PoolDetails<AccountId, ParametrizedCurve, Currencies> {
+	pub manager: AccountId,
+	pub curve: ParametrizedCurve,
+	pub bonded_currencies: Currencies,
+	pub state: PoolStatus<Locks>,
+	pub transferable: bool,
+}
+
+impl<AccountId, ParametrizedCurve, Currencies> PoolDetails<AccountId, ParametrizedCurve, Currencies>
+where
+	AccountId: PartialEq,
+{
+	pub fn new(
+		manager: AccountId,
+		curve: ParametrizedCurve,
+		bonded_currencies: Currencies,
+		transferable: bool,
+		state: PoolStatus<Locks>,
+	) -> Self {
+		Self {
+			manager,
+			curve,
+			bonded_currencies,
+			transferable,
+			state,
+		}
+	}
+
+	pub fn is_manager(&self, who: &AccountId) -> bool {
+		who == &self.manager
+	}
+
+	pub fn is_minting_authorized(&self, who: &AccountId) -> bool {
```

Comment:
`is_authorized_to_mint` would be less confusing.

Reply:
I keep thinking these methods should just be `can_x` (`can_mint`, `can_burn`) etc. Short & sweet & simple.

Reply:
now called `can_mint`